Configurations/descrip.mms.tmpl

```diff
@@ -748,12 +751,19 @@ reconfigure reconf :
 
   sub generatesrc {
       my %args = @_;
-      (my $target = $args{src}) =~ s/\.[sS]$/.asm/;
       my $generator = join(" ", @{$args{generator}});
       my $generator_incs = join("", map { ' "-I'.$_.'"' } @{$args{generator_incs}});
       my $deps = join(", -\n\t\t", @{$args{generator_deps}}, @{$args{deps}});
 
-      if ($target !~ /\.asm$/) {
+      if ($args{src} =~ /\.ld$/) {
```

Comment:
I'd argue that it would be more than appropriate to re-model this if-else chain. Because it's now switch-like with 3 options. It now looks `if (a) {} elsif {!b) {} else {}`, and this is hardly readable and becomes unmanageable if one attempts to add 4th option. It should be rewritten as `if (a) {} elsif (b) {} else {}`. Remark naturally applies to other makefile templates.